pkg/controllers/dnspolicy/dnspolicy_controller.go

```
@@ -154,19 +157,27 @@ func (r *DNSPolicyReconciler) reconcileResources(ctx context.Context, dnsPolicy
 		return err
 	}
 
-	if err := r.reconcileDNSRecords(ctx, dnsPolicy, gatewayDiffObj); err != nil {
+	if err = r.reconcileDNSRecords(ctx, dnsPolicy, gatewayDiffObj); err != nil {
+		gatewayCondition = buildGatewayCondition(conditions.DNSPolicyReasonInvalid, err)
+		_ = r.updateGatewayCondition(ctx, gatewayCondition, gatewayDiffObj)
```

not sure why we are ignoring the error here?